pkg/utils/etcdutil/testutil.go

```diff
@@ -50,3 +53,95 @@ func NewTestSingleConfig(t *testing.T) *embed.Config {
 func genRandName() string {
 	return "test_etcd_" + strconv.FormatInt(time.Now().UnixNano()%10000, 10)
 }
+
+// NewTestEtcdCluster is used to create a etcd cluster for the unit test purpose.
+func NewTestEtcdCluster(t *testing.T, count int) (servers []*embed.Etcd, etcdClient *clientv3.Client, clean func()) {
+	re := require.New(t)
+	servers = make([]*embed.Etcd, 0, count)
+
+	cfg := newTestSingleConfig(t)
+	etcd, err := embed.StartEtcd(cfg)
+	re.NoError(err)
+	etcdClient, err = CreateEtcdClient(nil, cfg.LCUrls)
+	re.NoError(err)
+	<-etcd.Server.ReadyNotify()
+	servers = append(servers, etcd)
+
+	for i := 1; i < count; i++ {
+		// Check the client can get the new member.
+		listResp, err := ListEtcdMembers(etcdClient)
+		re.NoError(err)
+		re.Len(listResp.Members, i)
+		// Add a new member.
+		etcd2 := MustAddEtcdMember(t, cfg, etcdClient)
+		cfg2 := etcd2.Config()
+		cfg = &cfg2
+		<-etcd2.Server.ReadyNotify()
+		servers = append(servers, etcd2)
+	}
+
+	checkMembers(re, etcdClient, servers)
+
+	clean = func() {
+		etcdClient.Close()
+		for _, server := range servers {
+			if server != nil {
+				server.Close()
+			}
+		}
+	}
+
+	return
+}
+
+// MustAddEtcdMember is used to add a new etcd member to the cluster.
+func MustAddEtcdMember(t *testing.T, cfg1 *embed.Config, client *clientv3.Client) *embed.Etcd {
```

Comment:
Why call it Mustxxx?

Reply:
Do you have any good ideas about the name of this test function?

Reply:
Here is no guarantee that the member is added.

Reply:
It will retry three times, and if it fails all three times, it will report an error `re.NoError(err, "addEtcdMemberWithRetry failed after retry")`

Reply:
I use another way to make the test stable, which is to wait member number to be normal before adding a new node.